### PR TITLE
group clusterOptions flags

### PR DESCRIPTION
**Pull request by raymond-zhang00**
*Issue #, if available:*

*Description of changes:*

- group clusterOptions flags
- add checkTinkerbellFlags function

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/flags.go [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+
+func addBundlesOverrideFlag(cmd *cobra.Command, path *string) {
+	cmd.Flags().StringVar(path, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+}
+
+func addForceCleanupFlag(cmd *cobra.Command, path *bool) {
+	cmd.Flags().BoolVar(path, "force-cleanup", false, "Force deletion of previously created bootstrap cluster")
+}
+
+func setupClusterOptionFlags(cmd *cobra.Command, clusterOpt *clusterOptions) {
```
Comment (raymond-zhang00) [inline cmd/eksctl-anywhere/cmd/flags.go:38]: Think this is a natural grouping for ClusterOption flags - relevant for `create` and `upgrade`. `validate` doesn't use all the flags and `delete` has a slight modification in usage. 

I don't know if it's worth complicating the func with overrides to accommodate those commands.

File: cmd/eksctl-anywhere/cmd/flags.go [hunk 2]
```diff
@@ -16,3 +20,39 @@ func bindFlagsToViper(cmd *cobra.Command, args []string) error {
...
 	return err
 }
+
+func setupClusterOptionFlags(cmd *cobra.Command, clusterOpt *clusterOptions) {
+	cmd.Flags().StringVarP(&clusterOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	cmd.Flags().StringVar(&clusterOpt.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+	cmd.Flags().StringVar(&clusterOpt.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
+}
+
+func addTinkerbellFlag(cmd *cobra.Command, path *string) {
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/flags.go:30]:
```suggestion
func applyTinkerbellHardwareFlag(cmd *cobra.Command, pathOut *string) {
```

File: cmd/eksctl-anywhere/cmd/upgradecluster.go [hunk 3]
```diff
@@ -77,16 +68,9 @@ func (uc *upgradeClusterOptions) upgradeCluster(cmd *cobra.Command) error {
...
-		// If no flag was returned there is a developer error as the flag has been removed
-		// from the program rendering it invalid.
-		if flag == nil {
-			panic("'hardwarefile' flag not configured")
-		}
-
-		if len(uc.hardwareCSVPath) != 0 && !validations.FileExists(uc.hardwareCSVPath) {
-			return fmt.Errorf("hardware config file %s does not exist", uc.hardwareCSVPath)
+		err := checkTinkerbellFlags(cmd, uc.hardwareCSVPath)
+		if err != nil {
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/upgradecluster.go:72]:
```suggestion
		if err := checkTinkerbellFlags(cmd, uc.hardwareCSVPath); err != nil {
```

File: cmd/eksctl-anywhere/cmd/flags.go [hunk 4]
```diff
@@ -16,3 +20,39 @@ func bindFlagsToViper(cmd *cobra.Command, args []string) error {
 	})
 	return err
 }
+
+func setupClusterOptionFlags(cmd *cobra.Command, clusterOpt *clusterOptions) {
+	cmd.Flags().StringVarP(&clusterOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	cmd.Flags().StringVar(&clusterOpt.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
+	cmd.Flags().StringVar(&clusterOpt.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
+}
+
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/flags.go:29]: Would it be appropriate to move the constants (`Tinkerbell*` etc) here too given these funcs seem to be the primary consumers now?

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 5]
```diff
@@ -79,20 +69,9 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
...
-		}
-
-		if !viper.IsSet(TinkerbellHardwareCSVFlagName) || viper.GetString(TinkerbellHardwareCSVFlagName) == "" {
-			return fmt.Errorf("required flag \"%v\" not set", TinkerbellHardwareCSVFlagName)
-		}
-
-		if !validations.FileExists(cc.hardwareCSVPath) {
-			return fmt.Errorf("hardware config file %s does not exist", cc.hardwareCSVPath)
+		err := checkTinkerbellFlags(cmd, cc.hardwareCSVPath)
+		if err != nil {
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/createcluster.go:73]:
```suggestion
		if err := checkTinkerbellFlags(cmd, cc.hardwareCSVPath); err != nil {
```

**Review by chrisdoherty4 — CHANGES_REQUESTED** [review]
Thanks for the change. Would you mind passing the `FlagSet` to `apply*Flags` funcs instead of the whole command object?

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 6]
```diff
@@ -43,23 +42,14 @@ var createClusterCmd = &cobra.Command{
...
-	createClusterCmd.Flags().StringVarP(
-		&cc.hardwareCSVPath,
-		TinkerbellHardwareCSVFlagName,
-		TinkerbellHardwareCSVFlagAlias,
-		"",
-		TinkerbellHardwareCSVFlagDescription,
-	)
+	setupClusterOptionFlags(createClusterCmd, &cc.clusterOptions)
+	setupTimeoutFlags(createClusterCmd, &cc.timeoutOptions)
+	addTinkerbellFlag(createClusterCmd, &cc.hardwareCSVPath)
```
Comment (g-gaston) [inline cmd/eksctl-anywhere/cmd/createcluster.go:47]: I don't have a strong preference on the verb, but I would suggest using the same one in all methods

**Review by g-gaston — APPROVED** [review]
Aside from naming, ltgm
Leaving the final approve for @chrisdoherty4

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 7]
```diff
@@ -43,23 +42,14 @@ var createClusterCmd = &cobra.Command{
...
-	createClusterCmd.Flags().StringVarP(&cc.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
-	createClusterCmd.Flags().StringVarP(
-		&cc.hardwareCSVPath,
-		TinkerbellHardwareCSVFlagName,
-		TinkerbellHardwareCSVFlagAlias,
-		"",
-		TinkerbellHardwareCSVFlagDescription,
-	)
+	applyClusterOptionFlags(createClusterCmd.Flags(), &cc.clusterOptions)
+	applyTimeoutFlags(createClusterCmd, &cc.timeoutOptions)
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/createcluster.go:46]: Any reason this still accepts the command?

Comment (raymond-zhang00) [reply cmd/eksctl-anywhere/cmd/createcluster.go]: I totally overlooked this one in cmd/options.go. Updated to accept FlagSet.

**Review by chrisdoherty4 — APPROVED** [review]
lgtm

**Pull request merged**